### ISSUE TITLE
API logging error objects

### ISSUE THREAD

**Issue by roberttod** [QubitProducts/driftwood#32]
I would like to be able to pass a second argument to `log.error` which would format and log an error object along with the message.

e.g.

``` js
const log = require('./log')
try {
  fn()
} catch (e) {
  log.error('fn threw an error', e)
}
```

**Comment by oliverwoodings** [comment]
👍  would love this. What if we made it support any arg being an error object? e.g. what if i just did:

```
log.error(e)
```

I'd kinda still want it to print the error even though its the first arg.

**Comment by roberttod** [comment]
Yeh, so how about we do the same as `console.log` and separate all the args with a space. Detect any error objects and log them in some special way.

**Comment by oliverwoodings** [comment]
Sounds delicious

**Comment by oliverwoodings** [comment]
The strategy i'm going to go for is that the right most arg can be a special object. The rest of the args will be joined into a single string, delimited by spaces.

**Comment by oliverwoodings** [comment]
![screen shot 2016-09-18 at 17 57 47](https://cloud.githubusercontent.com/assets/621323/18617527/7bee1b14-7dc9-11e6-9fe9-3676d5a2ba29.png)

**Comment by oliverwoodings** [comment]
Fixed by #27
